File: Leetcode/JS/Leetcode686-重复叠加字符串匹配.js
var repeatedStringMatch = function(a, b) {
    let s = a,i = 1;
    while(s.length < b.length){
        s += a;
        i++;
    }
    if(s.indexOf(b) != -1) return i;
    else{
        s += a;
        i++;
        if(s.indexOf(b) != -1) return i;
    }
    return -1;
};

File: Leetcode/sbgo/Leetcode686-重复叠加字符串匹配.go
package sbgo

import "strings"

func repeatedStringMatch(a string, b string) int {
	s, i := a, 1
	for len(s) < len(b) {
		s += a
		i++
	}
	if strings.Contains(s, b) {
		return i
	} else {
		s += a
		i++
		if strings.Contains(s, b) {
			return i
		}
	}
	return -1
}
